File: enggo-backend/src/main/java/com/nhom12/enggo_backend/dto/request/exam/RandomBlueprintRequest.java
package com.nhom12.enggo_backend.dto.request.exam;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class RandomBlueprintRequest {
    Byte difficulty;
    List<String> questionTypes = new ArrayList<>(List.of("MULTIPLE_CHOICE", "FILL_BLANK", "MATCHING"));
    List<Integer> themeIds;
    int totalQuestions;
}

File: AppEngGo/app/src/main/java/com/example/appenggo/model/RandomBlueprintRequest.kt
package com.example.appenggo.model

import com.google.gson.annotations.SerializedName

/**
 * Mirrors backend RandomBlueprintRequest used for generating a PvP quiz.
 */
 data class RandomBlueprintRequest(
    @SerializedName("difficulty")
    val difficulty: Byte,

    @SerializedName("questionTypes")
    val questionTypes: List<String> = listOf("MULTIPLE_CHOICE", "FILL_BLANK", "MATCHING"),

    @SerializedName("themeIds")
    val themeIds: List<Int>,

    @SerializedName("totalQuestions")
    val totalQuestions: Int
)
